majilove/models.py

```diff
@@ -106,3 +108,402 @@ def display_blood_type(self):
 
     def __unicode__(self):
         return unicode(self.t_name)
+
+############################################################
+# Photos
+
+class Photo(MagiModel):
+    collection_name = 'photo'
+    owner = models.ForeignKey(User, related_name='added_photos')
+    id = models.PositiveIntegerField(_('ID'), unique=True, primary_key=True, db_index=True)
+
+    name = models.CharField(_('Photo Name'), max_length=100)
+    japanese_name = models.CharField(string_concat(_('Photo Name'), ' (', _('Japanese'), ')'), max_length=100)
+    NAMES_CHOICES = LANGUAGES_NEED_OWN_NAME
+    d_names = models.TextField(_('Title'), null=True)
+    @property
+    def t_name(Self):
+        if get_language() == 'ja':
+            return self.japanese_name
+        return self.names.get(get_language(), self.name)
+
+    release_date = models.DateField(_('Release date'), null=True, db_index=True)
+    idol = models.ForeignKey(Idol, verbose_name=_('Idol'), related_name='photos', null=True, on_delete=models.SET_NULL, db_index=True)
+
+    # Images
+
+    # The square icon
+    icon = models.ImageField(_('Icon'), upload_to=uploadItem('p'), null=True)
+    icon_special_shot = models.ImageField(string_concat(_('Icon'), ' (', _('Special Shot'), ')'), upload_to=uploadItem('p/a'), null=True)
+
+    # Full photo
+    image = models.ImageField(_('Photo Image'), upload_to=uploadItem('p/image'))
+    image_special_shot = models.ImageField(string_concat(_('Photo Image'), ' (', _('Special Short'), ')'), upload_to=uploadItem('p/image/a'), null=True)
+
+    transparent = models.ImageField(_('Transparent'), upload_to=uploadItem('p/transparent'), null=True)
+    transparent_special_shot = models.ImageField(string_concat(_('Transparent'), ' (', _('Special shot'), ')'), upload_to=uploadItem('p/transparent/a'), null=True)
+
+    poster = models.ImageField(_('Poster'), upload_to=uploadItem('p/poster'), null=True)
+    poster_special_shot = models.ImageField(string_concat(_('Poster'), ' (', _('Special Shot'), ')'), upload_to=uploadItem('p/poster/a'), null=True)
+
+    message = models.ImageField(_('Message'), upload_to=uploadItem('p/message'), null=True)
+    autograph = models.ImageField(_('Autograph'), upload_to=uploadItem('p/autograph'), null=True)
+
+    message_text = models.TextField(string_concat(_('Message Text'), ' (', _('Japanese') + ')'), max_length=500, null=True)
+    message_translation = models.TextField(_('Message translation'), max_length=500, null=True)
+    MESSAGE_TRANSLATIONs_CHOICES = ALL_ALT_LANGUAGES
+    d_message_translations = models.TextField(_('Message translation'), null=True)
+    @property
+    def t_message_translation(self):
+        if get_language() == 'ja': return None
+        return self.message_translations.get(get_language(), self.message_translation)
```

Comment:
In this case there's no official English translation of the text (It's a message that gets written on a photo, so it's only in the game as an image, not as text), so I think it's better to store the japanese one separately.

Reply:
You can still store a translation without having an English value!

The form will show up like that:

<img width="713" alt="screen shot 2018-06-14 at 3 34 44 pm" src="https://user-images.githubusercontent.com/722281/41398003-865b6ecc-6fe8-11e8-9452-40591b9de6d4.png">

instead of the usual:

<img width="653" alt="screen shot 2018-06-14 at 3 34 50 pm" src="https://user-images.githubusercontent.com/722281/41398011-8c684146-6fe8-11e8-9cfb-73a046121f39.png">